### PR TITLE
`azurerm_kubernetes_cluster` - add support for `node_os_channel_upgrade`

### PR DESCRIPTION
**Pull request by stephybun**
Tests are still running.

Superseded #21386 
Related to #20171

### CODE REVIEW

**Review by mbfrahry — APPROVED** [review]
LGTM!

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -1459,6 +1471,21 @@ func resourceKubernetesClusterCreate(d *pluginsdk.ResourceData, meta interface{}
 		return err
 	}
 
+	autoUpgradeProfile := &managedclusters.ManagedClusterAutoUpgradeProfile{}
+	autoChannelUpgrade := d.Get("automatic_channel_upgrade").(string)
+	nodeOsChannelUpgrade := d.Get("node_os_channel_upgrade").(string)
+	if autoChannelUpgrade != "" {
+		if autoChannelUpgrade == string(managedclusters.UpgradeChannelNodeNegativeimage) && nodeOsChannelUpgrade != string(managedclusters.NodeOSUpgradeChannelNodeImage) {
```
Comment (mbfrahry) [inline internal/services/containers/kubernetes_cluster_resource.go:1478]: Looks like this wants to be 
```suggestion
		if autoChannelUpgrade == string(managedclusters.UpgradeChannelNodeNegativeimage) && string(managedclusters.NodeOSUpgradeChannelNodeImage) != "" && nodeOsChannelUpgrade != string(managedclusters.NodeOSUpgradeChannelNodeImage) {
```

to get around 
```
=== CONT  TestAccKubernetesCluster_upgradeChannel
testcase.go:110: Step 5/10 error: Error running apply: exit status 1
Error: `node_os_channel_upgrade` must be set to `NodeImage` if `automatic_channel_upgrade` is set to `node-image`
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 22, in resource "azurerm_kubernetes_cluster" "test":
22: resource "azurerm_kubernetes_cluster" "test" {
--- FAIL: TestAccKubernetesCluster_upgradeChannel (795.62s)
```

**Review by mbfrahry — CHANGES_REQUESTED** [review]
JK, I take it back. Almost there with a minor addition to fix an existing test

**Comment by stephybun** [comment]
Thanks for the review @mbfrahry! Relevant tests are passing so this should be good to merge.

**Comment by stevehipwell** [comment]
@stephybun @mbfrahry this is a breaking change for anyone who was setting `automatic_channel_upgrade` to `node-image` as if they upgrade the provider version to `v3.6.0` or greater the once working code will fail. If you really can't default `node_os_channel_upgrade` to `NodeImage` in the case where it's not explicitly set (I have no idea why not) then this code change should have triggered a SemVer major release.

Ideally this can be fixed in a patch release ASAP?

**Pull request merged**